src/auth/ejabberd_auth_anonymous.erl

```diff
@@ -317,18 +317,33 @@ remove_user(_LUser, _LServer) ->
 
 
 -spec supports_sasl_module(jid:lserver(), cyrsasl:sasl_module()) -> boolean().
-supports_sasl_module(Host, cyrsasl_anonymous) -> is_sasl_anonymous_enabled(Host);
-supports_sasl_module(Host, cyrsasl_plain) -> is_login_anonymous_enabled(Host);
-supports_sasl_module(Host, cyrsasl_scram) -> is_login_anonymous_enabled(Host);
-supports_sasl_module(Host, cyrsasl_scram_sha256) -> is_login_anonymous_enabled(Host);
-supports_sasl_module(Host, cyrsasl_digest) -> is_login_anonymous_enabled(Host);
-supports_sasl_module(_, _) -> false.
+supports_sasl_module(Host, cyrsasl_anonymous) ->
+    is_sasl_anonymous_enabled(Host);
+supports_sasl_module(Host, cyrsasl_plain) ->
+    is_login_anonymous_enabled(Host);
+supports_sasl_module(Host, cyrsasl_scram_sha1) ->
+    is_login_anonymous_enabled(Host) andalso
+    mongoose_scram:can_login_with_configured_password_format(Host, cyrsasl_scram_sha1);
```

Comment:
How the following here in other `cyrsasl_scram_sha*` clauses?
```suggestion
supports_sasl_module(Host, cyrsasl_scram_sha1 = ScramSha) ->
    is_login_anonymous_enabled(Host) andalso
    mongoose_scram:can_login_with_configured_password_format(Host, ScramSha);
```

The amount of lines is still the same, but we reduce the risk of copy-paste error :)

Reply:
Sure, good idea. Will update that in all auth backend modules.